qutebrowser/config/configtypes.py

```diff
@@ -303,26 +303,58 @@ class List(BaseType):
 
     """Base class for a (string-)list setting."""
 
-    def __init__(self, none_ok=False, valid_values=None):
+    def __init__(self, inner_type, none_ok=False):
         super().__init__(none_ok)
-        self.valid_values = valid_values
+        self.inner_type = inner_type
 
     def transform(self, value):
         if not value:
             return None
         else:
-            return [v if v else None for v in value.split(',')]
+            return [self.inner_type.transform(v.strip())
+                    for v in value.split(',')]
 
     def validate(self, value):
         self._basic_validation(value)
         if not value:
             return
-        vals = self.transform(value)
-        if None in vals:
-            raise configexc.ValidationError(value, "items may not be empty!")
+        for val in value.split(','):
+            self.inner_type.validate(val.strip())
+
+
+class BaseList(List):
```

Comment:
Couldn't you just use `typ.List(typ.String())` or `typ.List(typ.BaseType())` instead?